src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -0,0 +1,195 @@
+                                  rptId,
+                                  p.paymentToken()))
+                  : nodoOperations
+                      .activatePaymentRequest(partialPaymentRequestInfo, newTransactionRequestDto)
+                      .doOnSuccess(
+                          p ->
+                              log.info(
+                                  "Nodo activation for {} with paymentToken {}",
+                                  rptId,
+                                  p.paymentToken()));
+            })
+        .doOnNext(
+            paymentRequestInfo -> {
+              log.info(
+                  "Cache Nodo activation info for {} with paymentToken {}",
+                  rptId,
+                  paymentRequestInfo.paymentToken());
+              paymentRequestsInfoRepository.save(paymentRequestInfo);
+            })
+        .flatMap(
+            paymentRequestInfo -> {
+              final String transactionId = UUID.randomUUID().toString();
+              final SessionRequestDto sessionRequest =
+                  new SessionRequestDto()
+                      .email(newTransactionRequestDto.getEmail())
+                      .rptId(paymentRequestInfo.id().value())
+                      .transactionId(transactionId)
+                      .paymentToken(paymentRequestInfo.paymentToken());
+
+              return ecommerceSessionsClient
+                  .createSessionToken(sessionRequest)
+                  .map(sessionData -> Tuples.of(sessionData, paymentRequestInfo));
+            })
+        .flatMap(
+            args -> {
+              final SessionDataDto sessionDataDto = args.getT1();
+              final PaymentRequestInfo paymentRequestInfo = args.getT2();
+              final String paymentToken = paymentRequestInfo.paymentToken();
+              return isValidPaymentToken(paymentToken)
+                  ? Mono.just(
+                      Tuples.of(
+                          newTransactionActivatedEvent(
+                              paymentRequestInfo.amount(),
+                              paymentRequestInfo.description(),
+                              sessionDataDto.getEmail(),
+                              sessionDataDto.getTransactionId(),
+                              sessionDataDto.getRptId(),
+                              paymentToken),
+                          Mono.empty(),
+                          sessionDataDto))
+                  : Mono.just(
+                      Tuples.of(
+                          Mono.empty(),
+                          newTransactionActivationRequestedEvent(
+                              paymentRequestInfo.amount(),
+                              paymentRequestInfo.description(),
+                              sessionDataDto.getEmail(),
+                              sessionDataDto.getTransactionId(),
+                              sessionDataDto.getRptId()),
+                          sessionDataDto));
+            });
```

Comment:
Could be worth to refactor in the future to a `Mono<Tuple2<Either<TransactionActivatedEvent, TransactionActivateRequestedEvent>, SessionDataDto>>`